kmippp/kmippp.cpp

```diff
@@ -24,23 +24,27 @@ context::context(std::string server_address,
 
     if(SSL_CTX_use_certificate_file(ctx_, client_cert_fn.c_str(), SSL_FILETYPE_PEM) != 1)
     {
+        ERR_print_errors_fp(stderr);
```

Comment:
@lukin-oleksiy In my opinion using stderr in a dependent library is not a very good idea.
1. You may print to stderr without syncronization in parallel with another thread in mysqld.
2. Most probably, mysqld when run in deamon mode completely suppresses sederr output.
3. I would suggest to do something similar to https://github.com/percona/percona-server/blob/8.0/extra/opensslpp/src/opensslpp/core_error.cpp#L29 - basically extract OpenSSL error and throw an exception with this extracted message.

Reply:
fixed by adding custom exception